### DISCUSSION TITLE
Controllers from namespace outside of app

### DISCUSSION THREAD

**Discussion by danielsetreus** [spatie/laravel-route-attributes#68] [Ideas]
Our application setup is a bit... odd, for legacy reasons. In this example I'd like to load annotated controllers from a namespace outside of the application. In essense trying to define directories like

```php
'directories' => [
        app_path('Http/Controllers/Test'),
        '/home/parent/Http/Controllers/Test/',
]
```

where app_path gives a sub directory like `/home/yada/yada/wtf/`.

Problem is the namespace, as the package seems to try and determine the namespace based on its location, with the applications root namespace. So it would try and find controllers under `/home/parent` using namespace `App\Home\parent\Http\Controllers\Test\PhpTestController`. 

Would it be possible to, optionally, define directories with a pre-defined root namespace? Perhaps like

```php
'directories' => [
        app_path('Http/Controllers/Test'),
        [
            'dir' => '/home/parent/Http/Controllers/Test/',
            'rootNs' => 'ParentApp\Http\Controllers\Test',
        ],
]
```

**Comment by freekmurze** [comment]
I'd accept a PR that adds this if the code is simple enough. Make sure to add tests and update the readme.

**Reply by danielsetreus** [reply]
Opened #69 - and also feel really bad that I got such a great PR number :D

**Reply by freekmurze** [reply]
lol

**Comment by danielsetreus** [comment]
Featured merged in #69
